q2_types/per_sample_sequences/_transformer.py

```diff
@@ -26,11 +26,37 @@
 
 
 class PerSampleDNAIterators(dict):
-    pass
+    def __getitem__(self, key):
+        return super().__getitem__(key)()
+
+    def __iter__(self):
+        for sample_id, seqs in self.items():
+            yield sample_id, seqs()
 
 
 class PerSamplePairedDNAIterators(dict):
-    pass
+    def __getitem__(self, key):
+        fwd, rev = super().__getitem__(key)
+        return fwd(), rev()
+
+    def __iter__(self):
+        for sample_id, (fwd, rev) in self.items():
+            yield sample_id, (fwd(), rev())
+
+
+def _iterator_magic(filepath):
+    yield from skbio.io.read(filepath, format='fastq', constructor=skbio.DNA)
+
+
+def _prepare_manifest(dirfmt):
+    manifest = pd.read_csv(str(dirfmt.path / dirfmt.manifest.pathspec),
+                           header=0, comment='#').set_index('sample-id')
+    manifest.filename = manifest.filename.apply(lambda x: str(dirfmt.path / x))
+
+    manifest['data'] = manifest.filename.apply(
+        lambda filepath: lambda: _iterator_magic(filepath))
```

Comment:
This is like a triple η-abstraction with `_iterator_magic`. I'm pretty sure you just need the one...

```python
lambda filepath: skbio.io.read(filepath, format='fastq', constructor=skbio.DNA)
```
This will give you a series of Sequence generators (which you can return directly instead of the dataframe)